docs/user/glossary.rst

```diff
@@ -39,18 +40,15 @@ toggle additional features using the :guilabel:`Tools` menu for each term.
 
 .. image:: /images/glossary-tools.png
 
-Not translatable terms
-++++++++++++++++++++++
+Untranslatable terms
+++++++++++++++++++++
 
 .. versionadded:: 4.5
 
-Glossary terms which are read-only are not meant to be translated. You can use
-this for names or other terms which should not change while translating. Such
-terms are visually highlighted in the glossary sidebar.
-
-The terms can be flagged using :guilabel:`Tools` ↓
-:guilabel:`Mark as read-only`. In the background this toggles the ``read-only``
-flag of the string.
+Flagging certain glossary term translations ``read-only`` manually, or
+using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not
```

Comment:
Should is the imperative of the translator, they can not, because "read-only renders it impossible to do so.

Reply:
But it's up to translators whether they will follow whatever was defined in glossary or not... and also now it's up to translators whether to to define the flag or not...

Reply:
i.e. add the `read-only` flag to the glossary => the term shouldn't be translated in the translation.

Reply:
I think you are thinking of "forbidden" for the should. "read-only" is for sure "can't".

Adding an entry of the glossary as read-only would only prevent it from being changed, or translated _in the glossary_. Glossaries are just components now.
(I am so tired I wrote "preven't" X))

Reply:
I suppose you misunderstand the difference between `forbidden`, `not translatable` and *`read-only` for non-gloosary components*:
* `forbidden` is term with a given translation. And exactly this translation shouldn't be used e.g. `directory` shouldn't be translated as `folder`.
* `not translatable` is a term in glossary that shouldn't be translated in other translations at all. It is implemented via the `read-only` flag for a glossary, but unlike a normal component it could be removed or set by a translator. So `read-only` here is nothing more than an implementation notice. e.g. if `cat` (unix command) is excessively used through the text without any quotation you may want to add it as `not translatable`, to keep others from translating it as `cat` (small animal of felis genus). 
* *`read-only` for non-glossary* is what you describe here: a translation that can't be changed by a translator.

This section should describe the `not translatable` semantics at the first place (as it primary aimed for users). And it's backing implementation in the second. That's what actually was pretty straight forward in previous version.

Reply:
I was saying it makes sense to describe "forbidden" as should not, and "read-only" as can not. Seeing as one is a possible error to avoid, and the other is impossible to make.

We went from "not meant to be translated" and "should not change in translation" for _read-only_. That sounds a lot like change is possible to me.

Am I trying to describe "read-only" or "forbidden", and what am I doing wrong?

I don't particularly like the introduction of "not translatable", as a term, because it isn't a flag.

Reply:
> Am I trying to describe "read-only" or "forbidden", and what am I doing wrong?

> I don't particularly like the introduction of "not translatable", as a term, because it isn't a flag.

But "not translatable" just has a different semantics for glossaries, than "read-only" for normal components. It has to be introduced in some form.

All in all I believe it's time to call @nijel, and just ask to reimplement it via a new flag...

@nijel, long story short, implementation of not-translatable/untranslatable semantics through `read-only` flag looks quite confusing... Are there any reason not to do it via a separate flag?

Reply:
Why new flag? Take me through the current situation of what happens and what goes wrong.

If it has a function that is clearly described, ~ Can't be replicated in translation, and both strings and glossaries are components, what is the confusion?

Read-only means ~unchanged form only.
Forbidden means ~avoid this.

Reply:
Yes, glossaries are components now, but the `read-only` flag is treated differently and mean a different thing.

I've already wrote it above, as for now:

* `read-only` for non-glossaries: the translation of the string itself **can not** be changed by a user. (that's exactly, what you say, it's a strict **can not**)
* `read-only` for glossaries: the term is not-translatable, i.e. the term **should not** be (**is meant not to** be) changed while translating *other strings*, it's just makes it appear highlighted in the glossary sidebar, a user **can** (permitted, able and free to) translate it in other strings, but he **shouldn't**. As for the term itself, unlike non-glossaries a user **can** set and remove the `read-only` flag as he wish, which effectively allows him to translate the term itself.

Reply:
@comradekingu @Fat-Zer 
Guys, please close these endless conversations, this is hell to merge.

Reply:
@orangesunny, this conversation had stalled without a good solution... I should had probably file a separate issue on the problem  rather than try to call **nijel** directly...